src/services/fdp-storage/fdp-storage.utils.ts

```diff
@@ -27,18 +28,37 @@ function extractDappIdFromRegex(url: string, regex: RegExp): string | null {
   return null
 }
 
-export function dappUrlToId(url: string, beeUrl: string): DappId {
-  assertBeeUrl(beeUrl)
+function extractDappIdFromWhitelistedDapps(url: string): string | undefined {
+  const { dappId } = whitelistedDapps.find(({ url: dappUrl }) => url.startsWith(dappUrl)) || {}
 
-  let dappId = extractDappIdFromRegex(url, constructBzzRegex(beeUrl))
+  return dappId
+}
 
-  if (dappId) {
-    return dappId
-  }
+function isDappIdForged(dappId: string, url: string): boolean {
+  let isWhitelistedDappId = false
+
+  const matchingUrl = whitelistedDapps.some(({ url: currentUrl, dappId: currentDappId }) => {
+    if (currentDappId === dappId) {
+      isWhitelistedDappId = true
+
+      return url.startsWith(currentUrl)
+    }
+
+    return false
+  })
+
+  return isWhitelistedDappId && !matchingUrl
+}
+
+export function dappUrlToId(url: string, beeUrl: string): DappId {
+  assertBeeUrl(beeUrl)
 
-  dappId = extractDappIdFromRegex(url, constructSubdomainRegex(beeUrl))
+  const dappId =
+    extractDappIdFromWhitelistedDapps(url) ||
+    extractDappIdFromRegex(url, constructBzzRegex(beeUrl)) ||
+    extractDappIdFromRegex(url, constructSubdomainRegex(beeUrl))
 
-  if (dappId) {
+  if (dappId && !isDappIdForged(dappId, url)) {
```

Comment:
why do you need this check if `extractDappIdFromWhitelistedDapps` returns `undefined` if it is not in the allowed dapp list? also you check whitelisting here with `dappId` against the set from which you got `dappId`. I'm a bit confused.

Reply:
Somebody can forge the URL, because we don't own the ENS names that we use as dappIds in the `whitelistedDapps`. For example, if a malicious dapp is deployed to `{beeURL}/bzz/fairdrive`, then that dapp would be able to access the whole personal storage of the user.

Reply:
I see.
In this case I would suggest to buy available ENS names for all dApps on the whitelist and you won't have this problem for sure.